### PR TITLE
fix: refactor status command

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

I really do think that doing some form of factory object mapping to custom opts structure is total waste of time and doesn't fit much of purpose. Factory is good abstraction on it's own. Removed it from this command specifically.

Added support for name and location

Added common handling for services missing so we can now reuse it.

### CODE REVIEW

File: pkg/shared/contextutil/util.go [hunk 1]
```diff
@@ -81,12 +83,17 @@ func GetCurrentRegistryInstance(f *factory.Factory) (*registrymgmtclient.Registr
...
 	if err != nil {
 		return nil, err
 	}
 
-	currCtx, err := GetCurrentContext(svcContext, f.Localizer)
+	return GetRegistryForServiceConfig(currCtx, f)
+
+}
+
+func GetRegistryForServiceConfig(currCtx *servicecontext.ServiceConfig, f *factory.Factory) (*registrymgmtclient.Registry, error) {
```
Comment (rkpattnaik780) [inline pkg/shared/contextutil/util.go:95]: Add comment.

File: pkg/shared/contextutil/util.go [hunk 2]
```diff
@@ -50,16 +50,19 @@ func GetCurrentKafkaInstance(f *factory.Factory) (*kafkamgmtclient.KafkaRequest,
...
+	currCtx, err := GetCurrentContext(svcContext, f.Localizer)
 	if err != nil {
 		return nil, err
 	}
 
-	currCtx, err := GetCurrentContext(svcContext, f.Localizer)
+	return GetKafkaForServiceConfig(currCtx, f)
+}
+
+func GetKafkaForServiceConfig(currCtx *servicecontext.ServiceConfig, f *factory.Factory) (*kafkamgmtclient.KafkaRequest, error) {
```
Comment (rkpattnaik780) [inline pkg/shared/contextutil/util.go:61]: Add comment.

**Review by rkpattnaik780 — COMMENTED** [review]
Works great! I think we should still show status if a service is missing.

File: pkg/shared/contextutil/util.go (hunk 1 above)
Comment (wtrocki) [reply pkg/shared/contextutil/util.go:95]:
```suggestion
// GetRegistryForServiceConfig fetching registry data for provided service config
func GetRegistryForServiceConfig(currCtx *servicecontext.ServiceConfig, f *factory.Factory) (*registrymgmtclient.Registry, error) {
```

File: pkg/shared/contextutil/util.go (hunk 2 above)
Comment (wtrocki) [reply pkg/shared/contextutil/util.go:61]:
```suggestion
func GetKafkaForServiceConfig(currCtx *servicecontext.ServiceConfig, f *factory.Factory) (*kafkamgmtclient.KafkaRequest, error) {
```
```suggestion
// GetKafkaForServiceConfig fetching kafka service data using ServiceConfig
func GetKafkaForServiceConfig(currCtx *servicecontext.ServiceConfig, f *factory.Factory) (*kafkamgmtclient.KafkaRequest, error) {
```

**Pull request merged**